flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/PseudoRandomValueSelector.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.util;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.util.EnvironmentInformation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigInteger;
+import java.util.Random;
+import java.util.function.Function;
+
+/**
+ * Initializes the {@link Configuration} for particular {@link ConfigOption}s with random values if unset.
+ */
+@Internal
+class PseudoRandomValueSelector {
+	private final Function<Integer, Integer> randomValueSupplier;
+
+	private static final Logger LOG = LoggerFactory.getLogger(PseudoRandomValueSelector.class);
+
+	private PseudoRandomValueSelector(Function<Integer, Integer> randomValueSupplier) {
+		this.randomValueSupplier = randomValueSupplier;
+	}
+
+	public <T> void select(Configuration configuration, ConfigOption<T> option, T... alternatives) {
+		if (configuration.contains(option)) {
+			return;
+		}
+		final Integer choice = randomValueSupplier.apply(alternatives.length);
+		T value = alternatives[choice];
+		LOG.info("Randomly selected {} for {}", value, option.key());
+		configuration.set(option, value);
+	}
+
+	public static PseudoRandomValueSelector create(Object entryPointSeed) {
+		final long commitSeed = new BigInteger(EnvironmentInformation.getGitCommitId(), 16).hashCode();
+		final long combinedSeed = commitSeed << 32 | entryPointSeed.hashCode();
```

Comment:
So for the given commit id and given test name, the random selection choices will be always the same, on each run on any environment/machine?

If so, maybe put that (^^^) as a java doc for this `create` method?

Reply:
Yes that was my intent: have it reproducible without changing anything by just checking out the particular commit that failed on AZP.

I'm double-checking that `EnvironmentInformation.getGitCommitId()` works as expected or try a different approach to inject commit id.

Reply:
So I verified that randomized unaligned checkpoint configuration is actually picked up while creating the controllers.